src/bundles/peer-locations.test.js

```diff
@@ -267,3 +267,125 @@ it('should resolve alternative address for failed address lookup', async () => {
   expect(Object.keys(peerLocs)).toHaveLength(1)
   expectLocation(peerLocs[nextPeers[1].peer.toB58String()])
 })
+
+describe('getPublicIP', () => {
+  it('returns undefined on null identity', async () => {
+    expect(getPublicIP()).toEqual(undefined)
+    expect(getPublicIP(null)).toEqual(undefined)
+    expect(getPublicIP(undefined)).toEqual(undefined)
+  })
+
+  it('returns undefined on local IPs only', async () => {
+    const res = getPublicIP({
+      addresses: [
+        '/ip6/::1/tcp/4003',
+        '/ip4/127.0.0.1/tcp/4003',
+        '/ip4/192.168.0.0/tcp/4003',
+        '/ip4/192.168.255.255/tcp/4003',
+        '/ip4/10.0.0.0/tcp/4003',
+        '/ip4/10.255.255.255/tcp/4003',
+        '/ip4/172.16.0.0/tcp/4003',
+        '/ip4/172.16.255.255/tcp/4003'
+      ]
+    })
+
+    expect(res).toEqual(undefined)
+  })
+
+  it('returns undefined on discovery.libp2p.io', async () => {
+    const res = getPublicIP({
+      addresses: [
+        '/dnsaddr/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star/ipfs/QmbJbcN3Fvy5bC7Tr95STx5VFiP1G1WLPCHNceh1yShfbb',
+        '/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star/ipfs/QmYy3ka6HsQzpdTXY63nUKShsVdgg5zdhMXZXFeNGPyMT4'
+      ]
+    })
+
+    expect(res).toEqual(undefined)
+  })
+
+  it('returns correct IPv4', async () => {
+    const res = getPublicIP({
+      addresses: [
+        '/ip4/127.0.0.1/tcp/4003',
+        '/ip4/126.1.0.1/tcp/4003'
+      ]
+    })
+
+    expect(res).toEqual('126.1.0.1')
+  })
+
+  it('returns correct IPv6', async () => {
+    const res = getPublicIP({
+      addresses: [
+        '/ip4/127.0.0.1/tcp/4003',
```

Comment:
let's include localhost v6 too:
```suggestion
        '/ip4/127.0.0.1/tcp/4003',
        '/ip6/::1/tcp/4003',        
```